hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -706,30 +751,7 @@ ChangesetInfoPtr XmlChangeset::splitChangeset(ChangesetInfoPtr changeset)
   //  If there is only one element then splitting only marks that element as failed
   if (changeset->size() == 1)
   {
-    //  Iterate the three changeset type arrays looking for elements to mark
-    for (int current_type = ChangesetType::TypeCreate; current_type != ChangesetType::TypeMax; ++current_type)
-    {
-      for (ChangesetInfo::iterator it = changeset->begin(ElementType::Relation, (ChangesetType)current_type);
-           it != changeset->end(ElementType::Relation, (ChangesetType)current_type); ++it)
-      {
-        //  Set the relation's status to failed
-        _allRelations[*it]->setStatus(XmlElement::ElementStatus::Failed);
-        //  Update the failed count
-        ++_failedCount;
-      }
-      for (ChangesetInfo::iterator it = changeset->begin(ElementType::Way, (ChangesetType)current_type);
-           it != changeset->end(ElementType::Way, (ChangesetType)current_type); ++it)
-      {
-        _allWays[*it]->setStatus(XmlElement::ElementStatus::Failed);
-        ++_failedCount;
-      }
-      for (ChangesetInfo::iterator it = changeset->begin(ElementType::Node, (ChangesetType)current_type);
-           it != changeset->end(ElementType::Node, (ChangesetType)current_type); ++it)
-      {
-        _allNodes[*it]->setStatus(XmlElement::ElementStatus::Failed);
-        ++_failedCount;
-      }
-    }
+    updateFailedChangeset(changeset);
```

Comment:
Refactored this out into another function that sets the failed flag and increments the failed count only after a fix has failed.